ooniprobe/View/Informed consent/ICSwipeViewController.m

```diff
@@ -0,0 +1,122 @@
+#import "ICSwipeViewController.h"
+
+@interface ICSwipeViewController ()
+
+@end
+
+@implementation ICSwipeViewController 
+
+- (UIViewController *)first {
+    if (!_first) {
+        UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Onboarding" bundle:nil];
+        _first = [sb instantiateViewControllerWithIdentifier:@"Onboarding_1"];
+    }
+    return _first;
+}
+
+- (UIViewController *)second {
+    if (!_second) {
+        UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Onboarding" bundle:nil];
+        _second = [sb instantiateViewControllerWithIdentifier:@"Onboarding_2"];
+    }
+    return _second;
+}
+- (UIViewController *)third {
+    if (!_third) {
+        UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Onboarding" bundle:nil];
+        _third = [sb instantiateViewControllerWithIdentifier:@"Onboarding_3"];
+    }
+    return _third;
+}
+- (UIViewController *)fourth {
+    if (!_fourth) {
+        UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Onboarding" bundle:nil];
+        _fourth = [sb instantiateViewControllerWithIdentifier:@"Onboarding_4"];
+    }
+    return _fourth;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    question_number = 1;
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(nextPage) name:@"nextPage" object:nil];
+
+    self.PageViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"PageViewController"];
+    self.PageViewController.dataSource = self;
+    viewControllers = [[NSArray alloc] initWithObjects:self.first, self.second, self.third, self.fourth, nil];
```

Comment:
Objects are autocreated here because you reference the properties, right?

Reply:
yes